### PR TITLE
hero image slider

### PR DESCRIPTION
**Pull request by priscillaapaxton**
# PR Template

HeroImageSlider is ready to be imported and rendered in app component. for now it is just a component that exists, but isn't being rendered anywhere

- [] I have ran all tests (if applicable) and they are passing. 
(Haven't adjusted cypress tests. Will get to that once we've figured out the video)

### Types of changes made?

- [x]  javascript
- [x]  functionality

### What does this PR do?

- Adds functionality for hero image slideshow
- Updates JS for heroImageSlider component

### Relevant Tickets?

- Partially finishes iteration or ticket for [Welcome Page](https://github.com/The-Never-Ending-Story/front-end/issues/40) 

### Questions/Comments/Relevant Screenshots?
As images change, there is a split second where the image is loading and the background image/color will be visible. This is something we discussed in standup earlier and might be an issue that could be fixed with locally stored images or something that BE can address. Something to keep in mind. I also did not include the explore worlds/view random world in the hero image slideshow since I think we determined it would be in the about section? If we decide we do want those clickable links in the hero image carousel, they can be added easily.

### CODE REVIEW

**Review by sdmisra — APPROVED** [review]
Looks good! I think the height property of the slider-div class in CSS might be something to fiddle with to make sure it covers the viewport on phones and laptops, etc

The component itself looks good - I intend on putting it as the first thing to render upon welcomePage so it can sit in the background nicely.

File: src/Components/HeroImageSlider/HeroImageSlider.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      if(currentSlide === 11) {
+        setCurrentSlide(0)
+      } else {
+        setCurrentSlide(currentSlide+1)
+      }
+    }, 4000)
+    return ()=> clearTimeout(timer)
+  }, [currentSlide])
+  
+  if (slides.length === 0) {
```
Comment (Adam-Meza) [inline src/Components/HeroImageSlider/HeroImageSlider.js:27]: nice error handling

File: src/Components/HeroImageSlider/HeroImageSlider.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    return <LoadingIcon />
+  } else if (error) {
+    return <Error />
+  } else if (slides.length > 0) {
+    const heroImageStyle = {
+      backgroundImage: `url(${slides[currentSlide].img.landscape})`
+    }
+    return (
```
Comment (Adam-Meza) [inline src/Components/HeroImageSlider/HeroImageSlider.js:35]: simple , clean.

**Review by Adam-Meza — APPROVED** [review]
Functionality looks good. Lets figure out how and in what order were gonna be merging everything. Might wanna set aside more time than we think itll take

**Pull request merged**